README.pod
=pod

=head1 NAME

Dist::Zilla::App::Command::installdeps - Install author dependencies for a
Dist::Zilla dist

=head1 SYNOPSIS

  dzil installdeps [--install-command="cmd"] [--no-recommends] [--suggests]

=head1 DESCRIPTION

Installs all dependencies needed for building, testing, and releasing a
distribution managed by L<Dist::Zilla>. First authordeps needed to build the
distribution from F<dist.ini> are installed, then the distribution's
dependencies are installed, including the develop phase. Similar to running:

  dzil authordeps --missing | cpanm
  dzil listdeps --missing --author --cpanm-versions | cpanm

=head1 OPTIONS

=head2 --install-command

Command to run to install dependencies. Dependencies will be appended to the
end of the command in the L<format expected by cpanm|cpanm/"(arguments)">.
Defaults to just C<cpanm>.

=head2 --recommends / --no-recommends

Install recommended dependencies (or don't). Defaults to on.

=head2 --suggests / --no-suggests

Install suggested dependencies (or don't). Defaults to off.

=head1 BUGS

Report any issues on the public bugtracker.

=head1 AUTHOR

Dan Book <dbook@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2017 by Dan Book.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)

=head1 SEE ALSO

L<Dist::Zilla::App::Command::authordeps>,
L<Dist::Zilla::App::Command::listdeps>, L<Dist::Zilla::App::Command::stale>

=cut


Changes
{{$NEXT}}

0.001     2017-08-04 20:55:01 EDT
  - First release


Build.PL
# This Build.PL for Dist-Zilla-App-Command-installdeps was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.025
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.034',
);

my %errors = map {
    eval "require $_; $_->VERSION($configure_requires{$_}); 1";
    $_ => $@,
} keys %configure_requires;

if (!grep { $_ } values %errors)
{
    # This section for Dist-Zilla-App-Command-installdeps was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.015.
    use strict;
    use warnings;

    use 5.010001;
    # use Module::Build::Tiny 0.034;
    Module::Build::Tiny::Build_PL();
}
else
{
    if (not $ENV{PERL_MB_FALLBACK_SILENCE_WARNING})
    {
        warn <<'EOW'
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.


Errors from configure prereqs:
EOW
        . do {
            require Data::Dumper; Data::Dumper->new([ \%errors ])->Indent(2)->Terse(1)->Sortkeys(1)->Dump;
        };

        sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
    }


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v6.010.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "configure_requires" => {
        "Module::Build::Tiny" => "0.034"
      },
      "dist_abstract" => "Install author dependencies for a Dist::Zilla dist",
      "dist_author" => [
        "Dan Book <dbook\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-App-Command-installdeps",
      "dist_version" => "0.002",
      "license" => "artistic_2",
      "module_name" => "Dist::Zilla::App::Command::installdeps",
      "recursive_test_files" => 1,
      "requires" => {
        "CPAN::Meta::Requirements" => 0,
        "Dist::Zilla::App" => 0,
        "Dist::Zilla::Path" => 0,
        "Dist::Zilla::Util::AuthorDeps" => 0,
        "Module::Runtime" => 0,
        "String::ShellQuote" => 0,
        "perl" => "5.010001"
      },
      "test_requires" => {
        "File::Spec" => 0,
        "Module::Metadata" => 0,
        "Test::More" => 0
      }
    );


    my %fallback_build_requires = (
      "File::Spec" => 0,
      "Module::Metadata" => 0,
      "Test::More" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);


    $build->create_build_script;
}


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.010.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Install author dependencies for a Dist::Zilla dist",
  "AUTHOR" => "Dan Book <dbook\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-App-Command-installdeps",
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "Dist::Zilla::App::Command::installdeps",
  "PREREQ_PM" => {
    "CPAN::Meta::Requirements" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Path" => 0,
    "Dist::Zilla::Util::AuthorDeps" => 0,
    "IPC::System::Simple" => 0,
    "Module::Runtime" => 0,
    "String::ShellQuote" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.001",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Meta::Requirements" => 0,
  "Dist::Zilla::App" => 0,
  "Dist::Zilla::Path" => 0,
  "Dist::Zilla::Util::AuthorDeps" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IPC::System::Simple" => 0,
  "Module::Runtime" => 0,
  "String::ShellQuote" => 0,
  "Test::More" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
